test/helpers/quictestlib.c

```diff
@@ -0,0 +1,743 @@
+
+static int packet_plain_mutate(const QUIC_PKT_HDR *hdrin,
+                               const OSSL_QTX_IOVEC *iovecin, size_t numin,
+                               QUIC_PKT_HDR **hdrout,
+                               const OSSL_QTX_IOVEC **iovecout,
+                               size_t *numout,
+                               void *arg)
+{
+    OSSL_QUIC_FAULT *fault = arg;
+    size_t i, bufsz = 0;
+    unsigned char *cur;
+
+    /* Coalesce our data into a single buffer */
+
+    /* First calculate required buffer size */
+    for (i = 0; i < numin; i++)
+        bufsz += iovecin[i].buf_len;
+
+    fault->pplainio.buf_len = bufsz;
+
+    /* Add an allowance for possible growth */
+    bufsz += GROWTH_ALLOWANCE;
+
+    fault->pplainio.buf = cur = OPENSSL_malloc(bufsz);
+    if (cur == NULL) {
+        fault->pplainio.buf_len = 0;
+        return 0;
+    }
+
+    fault->pplainbuf_alloc = bufsz;
+
+    /* Copy in the data from the input buffers */
+    for (i = 0; i < numin; i++) {
+        memcpy(cur, iovecin[i].buf, iovecin[i].buf_len);
+        cur += iovecin[i].buf_len;
+    }
+
+    fault->pplainhdr = *hdrin;
+
+    /* Cast below is safe because we allocated the buffer */
+    if (fault->pplaincb != NULL
+            && !fault->pplaincb(fault, &fault->pplainhdr,
+                                (unsigned char *)fault->pplainio.buf,
+                                fault->pplainio.buf_len, fault->pplaincbarg))
+        return 0;
+
+    *hdrout = &fault->pplainhdr;
+    *iovecout = &fault->pplainio;
+    *numout = 1;
+
+    return 1;
+}
+
+static void packet_plain_finish(void *arg)
+{
+    OSSL_QUIC_FAULT *fault = arg;
+
+    /* Cast below is safe because we allocated the buffer */
+    OPENSSL_free((unsigned char *)fault->pplainio.buf);
+    fault->pplainio.buf_len = 0;
+    fault->pplainbuf_alloc = 0;
+    fault->pplainio.buf = NULL;
+}
+
+int ossl_quic_fault_set_packet_plain_listener(OSSL_QUIC_FAULT *fault,
+                                              ossl_quic_fault_on_packet_plain_cb pplaincb,
+                                              void *pplaincbarg)
+{
+    fault->pplaincb = pplaincb;
+    fault->pplaincbarg = pplaincbarg;
+
+    return ossl_quic_tserver_set_plain_packet_mutator(fault->qtserv,
+                                                      packet_plain_mutate,
+                                                      packet_plain_finish,
+                                                      fault);
+}
+
+/* To be called from a packet_plain_listener callback */
+int ossl_quic_fault_resize_plain_packet(OSSL_QUIC_FAULT *fault, size_t newlen)
+{
+    unsigned char *buf;
+    size_t oldlen = fault->pplainio.buf_len;
+
+    /*
+     * Alloc'd size should always be non-zero, so if this fails we've been
+     * incorrectly called
+     */
+    if (fault->pplainbuf_alloc == 0)
+        return 0;
+
+    if (newlen > fault->pplainbuf_alloc) {
+        /* This exceeds our growth allowance. Fail */
+        return 0;
+    }
```

Comment:
Strictly speaking the `{}` are unnecessary.

Reply:
I would change our coding style to have this explicitly required. 
The extra braces improves readability IMHO.